### PR TITLE
milestone: Used Redux in React components

### PR DESCRIPTION
**Pull request by Roseokpe**
In this milestone I used Redux store to display books

### CODE REVIEW

File: src/components/NewBook.jsx [hunk 1]
```diff
@@ -1,22 +1,71 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
```
Comment (DammyShittu) [inline src/components/NewBook.jsx]: - [ ] You are using `uuid` for your unique id generator, but you have no installed the package. Please do that.

  uuid

File: src/components/CategoryPage.jsx [hunk 2]
```diff
@@ -1,9 +1,19 @@
...
+import {useSelector, useDispatch} from 'react-redux';
+import {checkStatus} from '../redux/categories/categories'
 
-const CategoryPage = () => (
-  <div>
-    <button type="button">CHECK STATUS</button>
-  </div>
-);
+const CategoryPage = () => {
+  const categories = useSelector((state) => state.checkReducer.categories);
```
Comment (DammyShittu) [inline src/components/CategoryPage.jsx]: - [ ] Nothing is being displayed on your categories page. If you visit the page and check your console, you will see this is due to some errors here. You need to change this line to the code below as what you have in your store is `category`, not `checkReducer.categories`.

  ```
  const categories = useSelector((state) => state.category);
  ```

- [ ] After that you need to refactor your categories file in the redux to this.

  ```
  const CHECK_STATUS = 'bookStore/categories/CHECK_STATUS';

  const categoriesState = [];
  export const checkStatus = () => ({
    type: CHECK_STATUS,
  });
  
  const checkReducer = (state = categoriesState, action) => {
    switch (action.type) {
      case CHECK_STATUS:
        return 'Under construction...';
      default:
        return state;
    }
  };
  
  export default checkReducer;
  ```

**Review by DammyShittu — COMMENTED** [review]
Hi @Roseokpe,

## Highlights
✔️ Descriptive PR
✔️ Passing linters

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [ ] Your books are being added to the store but they are displayed on the page. In your BooksPage component, you need to import `useSelector` from `react-redux` and restructure your bookList variable to this:

  ```
  const bookList = useSelector((state) => state.books)
  ```

- [ ] Books are not being removed when I click the remove button. Kindly look at your BookItem component and add a dispatch that removes the book from the store using the book id. Ensure the book is properly removed.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BertrandConxy — APPROVED** [review]
Hi @Roseokpe , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💪 
- No linter errors
- project requirements are met
- project works fine
- good PR description
- correct flow is used

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**